### ISSUE TITLE
No contributions dialog keeps popping repeatedly.

### ISSUE THREAD

**Issue by yashk2000** [commons-app/apps-android-commons#3412]
**Summary:** 

A dialog saying "You haven't, made any contributions yet" keeps showing up every time the achievements activity is opened when user has no contributions.

**What did you expect the app to do, and what did you see instead?**

The dialog should be shown, but the user should be given an option of "Do not show again". Else the dialog starts feeling a bit irritating every time the achievements activity is opened.

**Device and Android version:** 

Nokia 7 Plus, Android 10
 
**Commons app version:** 

2.12.3-debug-master

**Would you like to work on the issue?**

Yes.

**Comment by sivaraam** [comment]
Wouldn't it actually be better to handle the case of having no uploads in a different way? Instead of showing a pop-up stating the user has no contributions, I think it would be better to show some motivating content in the Achievements screen which would induce the user to kickstart his uploading journey. I don't have particular ideas about what could be done yet. But just wanted to share this thought.

**Comment by yashk2000** [comment]
It's a good idea :+1:
I'll try to think of what we can do without making it feel that we're imposing something on the user.
Until then I think we can continue with this.
As soon as I figure out an idea I'll comment it here.

**Comment by maskaravivek** [comment]
> Wouldn't it actually be better to handle the case of having no uploads in a different way? Instead of showing a pop-up stating the user has no contributions, I think it would be better to show some motivating content in the Achievements screen which would induce the user to kickstart his uploading journey. I don't have particular ideas about what could be done yet. But just wanted to share this thought.

**Option 1:** 
Yes, I think we can add a CTA with "Make your first upload" or something similar and trigger the upload flow. Our upload flow is easily pluggable and can be invoked from any place with just a couple of lines of codes. 

we can also add a "do not show again" button in addition to Cancel. So all in all we can have 3 buttons in the dialog
- Cancel: dialog dismissed, pops up again when the page is opened for the next time
- Do not show again: dismissed, never shown again
- Make your first upload: Invokes upload flow. 

**Option 2**
Or else, we can convert the dialog to a toast message so that it is less obstructive or replace the dialog with an empty screen text with "Make your first upload button".

**Comment by yashk2000** [comment]
@maskaravivek the first option sounds good. This way the user can directly make a new upload and it won't be obstructive as the user can simply choose not to see the dialog again after it is shown for the first time. 
Should I proceed with it?

**Comment by sivaraam** [comment]
@maskaravivek Your options are good and would be easy to implement which is a good thing as it allows us to do the change without much effort. That said, I wanted to mention that I was actually envisioning something more when I mentioned that we should show _motivating content_ in the Achievements screen which would induce the user to _kickstart his uploading journey_. I was envisioning something similar to what's elaborated in the Material design's [empty state design guideline](https://material.io/design/communication/empty-states.html) page. To give some practical examples, here are a couple of screenshots from other apps that show how they handle their empty states (no contributions in our case)

"Highlights" screen from the [Pocket](https://getpocket.com/) Android app:
![Screenshot_2020-02-16-20-05-07](https://user-images.githubusercontent.com/12448084/74607435-e7da9a80-50fe-11ea-9e76-edc5eef8829f.png)

Home screen of [Signal](https://signal.org/) app:
![Screenshot_2020-02-16-19-36-28](https://user-images.githubusercontent.com/12448084/74607472-22443780-50ff-11ea-9c34-6f4eb9b7bcd1.png)

I hope that gives an idea. I was actually thinking that we could remove the pop-up and show something like the ones above in the Achievements screen itself when the user has no contributions.

**Comment by maskaravivek** [comment]
@sivaraam Yes, that the 2nd option that I suggested. Option 1 would be easier and faster to implement but yes Option 2 would be aesthetically better. 

Let us get opinions from our designer @neslihanturan :)

**Comment by sivaraam** [comment]
> @sivaraam Yes, that the 2nd option that I suggested. Option 1 would be easier and faster to implement but yes Option 2 would be aesthetically better.
> 

Ah, you're right! I missed it initially as I got lost after reading the "... convert the dialog to a toast message so that it is less obstructive ..." part. My bad x-(

**Comment by yashk2000** [comment]
@maskaravivek @sivaraam making an empty screen with some content to motivate the user for uploading some content is good too and easy to implement as well. 

![Screenshot_20190511-142206](https://user-images.githubusercontent.com/41234408/57567590-3d0fb000-73f9-11e9-8cc9-aa79d520c050.png)

This is an empty activity that I designed for another application. We can maybe make something similar which follows the rest of the appearance of the app. and put a better message.

**Comment by misaochan** [comment]
I like the idea of converting the dialog to an empty activity. Both the Pocket and Signal app's screen looks good (the one without any pictures is probably easier to implement). Please no gigantic emojis though. ;)

**Comment by nicolas-raoul** [comment]
Here are a few possible illustrations: https://commons.wikimedia.org/wiki/Category:Lonely_people_in_art lol
